sympy/solvers/solvers.py

```diff
@@ -1846,6 +1862,57 @@ def _ok_syms(e, sort=False):
     return result
 
 
+def nc_solve(expr, symbol):
+    def linear(expr, symbol):
+        expr = simplify(expr)
+        if expr.is_Add:
+            return all(linear(arg, symbol) for arg in expr.args)
+        if expr.is_Mul:
+            s = [arg for arg in expr.args if arg.has(symbol)]
+            if len(s) > 1:
+                return False
+            elif s:
+                return linear(s[0], symbol)
+            else:
+                return True
+
+        if (not expr.has(symbol)) or (expr == symbol):
+            return True
+        else:
+            return False
+
+    if not linear(expr, symbol):
+        raise NotImplementedError(filldedent('''
+            Non-linear equations in noncommutative variables
+            are not supported''')
+            )
+    ind = 0
+    if expr.is_Add:
+        ind, expr = expr.as_independent(symbol)
+    if not (expr.is_Add):
+        left, expr = expr.as_independent(symbol, as_Add=False)
+        right = 1/symbol*expr
+        return [-1/left*ind*1/right]
```

Comment:
No, because the `if` block preceding it changes what `expr` is. For example, consider solving `expr = M*x+a` in `x`. It enters `if` on line 1890 since `expr.is_Add = True` which gives `ind = a` and `expr = M*x`. Now `expr.is_Add = False` so it enters the next if and the returned result is `-M**(-1)*a`, not 0.